clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -134,11 +147,17 @@ private SslEngineFactory instantiateSslEngineFactory(Map<String, Object> configs
             sslEngineFactory = Utils.newInstance(sslEngineFactoryClass);
         }
         sslEngineFactory.configure(configs);
-        this.sslEngineFactoryConfig = configs;
-        return sslEngineFactory;
+        boolean pass = false;
+        try {
+            validator.accept(sslEngineFactory);
```

Comment:
```configure``` and ```reconfigure``` has different checks for the new ssl engine factory and the new ssl engine factory should be closed automatically if the custom checks don't pass.